pennylane/dla/structure_constants.py

```diff
@@ -0,0 +1,114 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""A function to compute the adjoint representation of a Lie algebra"""
+from itertools import combinations
+
+import numpy as np
+
+from pennylane.pauli import PauliSentence
+
+
+def _all_commutators(ops):
+    commutators = {}
+    for (j, op1), (k, op2) in combinations(enumerate(ops), r=2):
+        res = op1.commutator(op2)
+        if res != PauliSentence({}):
+            commutators[(j, k)] = res
+
+    return commutators
+
+
+def structure_constants(dla):
+    r"""
+    Compute the structure constants that make up the adjoint representation of a DLA.
+
+    Given a DLA :math:`\{iG_1, iG_2, .. iG_d \}` of dimension :math:`d`,
+    the structure constants yield the decomposition of all commutators in terms of DLA elements,
+
+    .. math:: [i G_\alpha, i G_\beta] = \sum_{\gamma = 0}^{d-1} f^\gamma_{\alpha, \beta} iG_\gamma.
+
+    The adjoint representation :math:`\left(\text{ad}(iG_\gamma)\right)_{\alpha, \beta} = f^\gamma_{\alpha, \beta}` is given by those structure constants,
+    which can be computed via
+
+    .. math:: f^\gamma_{\alpha, \beta} = \frac{\text{tr}\left(i G_\gamma \cdot \left[i G_\alpha, i G_\beta \right] \right)}{\text{tr}\left( iG_\gamma iG_\gamma \right)}.
+
+    Note that this is just the projection of the commutator on the DLA element :math:`iG_\gamma` via the trace inner product.
+    The inputs are assumed to be orthogonal. However, we neither assume nor enforce normalization of the DLA elements
+    :math:`G_\alpha`, hence the normalization
+    factor :math:`\text{tr}\left( iG_\gamma iG_\gamma \right)` in the projection.
+
+    Args:
+        dla (List[PauliSentence]): The (dynamical) Lie algebra for which we want to compute its adjoint representation. DLAs can be generated by a set of generators via :func:`~lie_closure`
+
+    Returns:
+        matrix: The adjoint representation of shape ``(d, d, d)``.
+
+    **Example**
+
+    Let us generate the DLA of the transverse field Ising model using :func:`~lie_closure`.
+
+    >>> n = 2
+    >>> gens = [PauliSentence({PauliWord({i:"X", i+1:"X"}): 1.}) for i in range(n-1)]
+    >>> gens += [PauliSentence({PauliWord({i:"Z"}): 1.}) for i in range(n)]
+    >>> dla = qml.dla.lie_closure(gens)
+    >>> print(dla)
+    [1.0 * X(0) @ X(1), 1.0 * Z(0), 1.0 * Z(1), -1.0 * Y(0) @ X(1), -1.0 * X(0) @ Y(1), -1.0 * Y(0) @ Y(1)]
+
+    The dimension of the DLA is :math:`d = 6`. Hence, the structure constants have shape ``(6, 6, 6)``.
+
+    >>> adjoint_rep = qml.dla.structure_constants(dla)
+    >>> adjoint_rep.shape
+    (6, 6, 6)
+
+    The structure constants tell us the commutation relation between operators in the DLA via
+
+    .. math:: [i G_\alpha, i G_\beta] = \sum_{\gamma = 0}^{d-1} f^\gamma_{\alpha, \beta} iG_\gamma.
+
+    Let us confirm those with some examples. Take :math:`[iG_1, iG_3] = [iZ_0, -iY_0 X_1] = -i 2 X_0 X_1 = -i 2 G_0`, so
```

Comment:
We're showing mostly one here, really.
```suggestion
    Let us confirm those with an example. Take :math:`[iG_1, iG_3] = [iZ_0, -iY_0 X_1] = -i 2 X_0 X_1 = -i 2 G_0`, so
```